[SSL] Update certificate-and-hostname-priority.md (#12661)
cloudflare/cloudflare-docs: content/ssl/reference/certificate-and-hostname-priority.md

---

## Hostname priority (Cloudflare for SaaS)

When multiple proxied DNS records exist for a zone — usually with Cloudflare for SaaS — only one record can control the zone settings and associated origin server.

Cloudflare determines this priority in the following order (assuming each record exists and is proxied (orange-clouded)):

1.  **Exact hostname match**:

>     1.  [New custom hostname](/cloudflare-for-platforms/cloudflare-for-saas/start/getting-started/) (Belonging to a SaaS Provider)
>     2.  [Legacy custom hostname](/cloudflare-for-platforms/cloudflare-for-saas/reference/versioning/) (Belonging to a SaaS Provider)

Reviewer: ```suggestion
    1.  [New custom hostname](/cloudflare-for-platforms/cloudflare-for-saas/start/getting-started/) (belonging to a SaaS provider)
    2.  [Legacy custom hostname](/cloudflare-for-platforms/cloudflare-for-saas/reference/versioning/) (belonging to a SaaS provider)
```

    3.  [DNS](/dns/manage-dns-records/reference/proxied-dns-records/) (Belonging to the logical DNS zone)

2.  **Wildcard hostname match**:

    1.  DNS (Belonging to the logical DNS zone)

>     2.  New custom hostname (Belonging to a SaaS Provider)

Reviewer: ```suggestion
    2.  New custom hostname (belonging to a SaaS provider)
```

> If a hostname resource record is not proxied (gray-clouded) for a zone on Cloudflare, that zone’s settings are not applied and any settings configured at the associated origin are applied instead. This origin could be another zone on Cloudflare or any other server.

Reviewer: ```suggestion
If a hostname resource record is not proxied (gray-clouded) for a zone on Cloudflare, that zone's settings are not applied and any settings configured at the associated origin are applied instead. This origin could be another zone on Cloudflare or any other server.
```

### Example scenarios

#### Scenario 1

> Customer1 uses Cloudflare as authoritative DNS for the zone `shop.example.com`. Customer2 is a SaaS provider that creates and successfully [verifies the new custom hostname](/cloudflare-for-platforms/cloudflare-for-saas/domain-support/hostname-validation/) `shop.example.com`. Afterward, traffic starts routing over Customer2’s zone:

Reviewer: ```suggestion
Customer1 uses Cloudflare as authoritative DNS for the zone 'shop.example.com'. Customer2 is a SaaS provider that creates and successfully [verifies the new custom hostname](/cloudflare-for-platforms/cloudflare-for-saas/domain-support/hostname-validation/) 'shop.example.com'. Afterward, traffic starts routing over Customer2's zone:
```

- If Customer1 wants to regain control of their zone, Customer1 contacts Customer2 and requests them to delete the custom hostname record. Another possibility is to stop proxying (gray-cloud) the record.

> - If Customer1 is already proxying a new custom hostname for `www.example.com`, Customer2 creates and verifies `www.example.com` so traffic starts routing over Customer2’s zone. Since this new custom hostname is the last one validated, the new custom hostname on Customer1’s zone enters a _moved_ status.

Reviewer: ```suggestion
- If Customer1 is already proxying a new custom hostname for 'www.example.com', Customer2 creates and verifies 'www.example.com' so traffic starts routing over Customer2's zone. Since this new custom hostname is the last one validated, the new custom hostname on Customer1's zone enters a _moved_ status.
```

> - If Customer1 is already proxying a legacy custom hostname for `www.example.com` and Customer2 creates and verifies a new wildcard custom hostname for `*.example.com`, traffic is routed to Customer1’s zone while the `www.example.com` CNAME points to Customer1.

Reviewer: ```suggestion
- If Customer1 is already proxying a legacy custom hostname for 'www.example.com' and Customer2 creates and verifies a new wildcard custom hostname for '*.example.com', traffic is routed to Customer1’s zone while the 'www.example.com' CNAME points to Customer1.
```

#### Scenario 2

A customer has a [proxied](/dns/manage-dns-records/reference/proxied-dns-records/) DNS record for their domain. The customer's zone on Cloudflare is using a Free plan.

This customer is also using a SaaS provider that utilizes Cloudflare for SaaS. The SaaS provider is using a Cloudflare Enterprise plan.

> If the  provider is using a wildcard custom hostname, then the original customer's plan limits will take precedence over the  provider's plan limits (Cloudflare will treat the zone as a Free zone). To apply the Enterprise limits through Cloudflare for SaaS, the original customer's zone would need to either use a [DNS-only](/dns/manage-dns-records/reference/proxied-dns-records/) record or the SaaS provider would need to use an exact hostname match.

Reviewer: ```suggestion
If the provider is using a wildcard custom hostname, then the original customer's plan limits will take precedence over the provider's plan limits (Cloudflare will treat the zone as a Free zone). To apply the Enterprise limits through Cloudflare for SaaS, the original customer's zone would need to either use a [DNS-only](/dns/manage-dns-records/reference/proxied-dns-records/) record or the SaaS provider would need to use an exact hostname match.
```
